java-frontend/src/main/java/org/sonar/java/model/JSema.java

```diff
@@ -135,4 +138,8 @@ ITypeBinding resolveType(String name) {
   public void cleanupEnvironment() {
     ASTUtils.cleanupEnvironment(ast);
   }
+
+  public Set<String> undefinedTypes() {
```

Comment:
Maybe it's better to make field private and return LinkedHashSet here too, if you want to keep order

Reply:
It looks like this method is never used. Do we need it?

Reply:
It is used by the `JavaAstScanner`, at the end of the analysis.